src/EnergyPlus/PipeHeatTransfer.hh

```diff
@@ -311,116 +305,75 @@ namespace PipeHeatTransfer {
 			EnvironmentHeatLossRate( 0.0 ),
 			EnvHeatLossEnergy( 0.0 ),
 			VolumeFlowRate( 0.0 )
-		{}
-
-	};
 
-	// Object Data
-	extern Array1D< PipeHTData > PipeHT;
-	extern Array1D< PipeHeatTransferReport > PipeHTReport;
-
-	// Functions
-
-	void
-	SimPipesHeatTransfer(
-		int const EquipType,
-		std::string const & EquipName, // name of the Pipe Heat Transfer.
-		int & EqNum, // index in local derived types for external calling
-		bool const InitLoopEquip,
-		bool const FirstHVACIteration // component number
-	);
+		{}
 
-	//==============================================================================
+		static
+		std::shared_ptr< PlantComponent > factory( int objectType, std::string objectName );
```

Comment:
Eventually I think the indeces will become fully moot, at least for these plant component and other "simple" component applications.  But right now there are still things that want to "grab" the data structures of these components by index.  We will _not_ be arguing that right now, eventually it will also go away, but for right now, leaving them in an Array1D that is index-base-1 keeps the possibilities for diffs and other stuff down to a minimum.